Unable to build Docker container

# The Issue
I am unable to build and run a Raneto container based on the Dockerfile

```
λ docker build . -t myranteo
Sending build context to Docker daemon  159.8MB
Step 1/7 : FROM node
latest: Pulling from library/node
d960726af2be: Pull complete 
e8d62473a22d: Pull complete 
8962bc0fad55: Pull complete 
65d943ee54c1: Pull complete 
532f6f723709: Pull complete 
f8463f32765b: Pull complete 
5714599515d2: Pull complete 
0f0993710fe6: Pull complete 
d09a14b7c3cb: Pull complete 
Digest: sha256:e36cf1bb8719551220ba8c3ee1583881e79ad040803570e0849b00b8fe009153
Status: Downloaded newer image for node:latest
 ---> d1b3088a17b1
Step 2/7 : EXPOSE 3000
 ---> Running in 21e0e92a4e38
Removing intermediate container 21e0e92a4e38
 ---> 7787daedb185
Step 3/7 : RUN mkdir -p /usr/src/app
 ---> Running in b751e221ea43
Removing intermediate container b751e221ea43
 ---> af833e3f679d
Step 4/7 : WORKDIR /usr/src/app
 ---> Running in ec51e6bc04ea
Removing intermediate container ec51e6bc04ea
 ---> 29aabf0e027b
Step 5/7 : COPY . /usr/src/app
 ---> 68bc042cb0f4
Step 6/7 : RUN npm install --production &&      ./node_modules/gulp/bin/gulp.js
 ---> Running in 2eaca80d2bab
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'bootstrap-rtl@3.3.4',
npm WARN EBADENGINE   required: { node: '~0.10.1' },
npm WARN EBADENGINE   current: { node: 'v16.3.0', npm: '7.15.1' }
npm WARN EBADENGINE }

> raneto@0.17.0 postinstall
> gulp default

sh: 1: gulp: not found
npm notice 
npm notice New minor version of npm available! 7.15.1 -> 7.16.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.16.0>
npm notice Run `npm install -g npm@7.16.0` to update!
npm notice 
npm ERR! code 127
npm ERR! path /usr/src/app
npm ERR! command failed
npm ERR! command sh -c gulp default

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-06-09T21_03_16_159Z-debug.log
The command '/bin/sh -c npm install --production &&     ./node_modules/gulp/bin/gulp.js' returned a non-zero code: 127
```

# System
- OS - Arch Linux
- Linux - 5.12.9-arch1-1
- Architecture - x86_64
- Docker version - `Docker version 20.10.7, build f0df35096d`

# Considerations/thoughts to resolve
- Dockerfile should not copy `node_modules` to the build context
- `npm install --production` doesn't install dev dependencies however `gulp` is listed as a dev dependency but `gulp` is called from the Dockerfile

Based on the above, as a user I would prefer not to have to have node and npm installed on my system and not have to install dependencies locally before building a container.  
I would expect the container to install all dependencies needed to run the application in an isolated environment.

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,4 @@
+.git/
+node_modules/
+bin/
+test/

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,13 @@
 FROM node
 
 EXPOSE 3000
+ENV HOST 0.0.0.0
+ENV PORT 3000
 
-RUN mkdir -p /usr/src/app
 WORKDIR /usr/src/app
 COPY . /usr/src/app
 
-RUN npm install --production && \
-	./node_modules/gulp/bin/gulp.js
+RUN npm install --production
+RUN ./node_modules/gulp/bin/gulp.js
 
 CMD ["npm", "start"]

--- a/package.json
+++ b/package.json
@@ -75,7 +75,8 @@
     "sweetalert2": "10.16.7",
     "underscore": "1.13.1",
     "underscore.string": "3.3.5",
-    "validator": "13.6.0"
+    "validator": "13.6.0",
+    "gulp": "4.0.2"
   },
   "devDependencies": {
     "chai": "4.3.4",
@@ -84,7 +85,6 @@
     "eslint-plugin-import": "2.22.1",
     "eslint-plugin-node": "11.1.0",
     "eslint-plugin-promise": "4.3.1",
-    "gulp": "4.0.2",
     "gulp-shell": "0.8.0",
     "mocha": "8.4.0",
     "pm2": "4.5.6",

